compaction/compaction.cc

```diff
@@ -137,14 +138,22 @@ std::string_view to_string(compaction_type_options::scrub::quarantine_mode quara
 
 static api::timestamp_type get_max_purgeable_timestamp(const table_state& table_s, sstable_set::incremental_selector& selector,
         const std::unordered_set<shared_sstable>& compacting_set, const dht::decorated_key& dk, uint64_t& bloom_filter_checks,
-        const api::timestamp_type compacting_max_timestamp) {
+        const api::timestamp_type compacting_max_timestamp, is_shadowable is_shadowable) {
     if (!table_s.tombstone_gc_enabled()) [[unlikely]] {
         return api::min_timestamp;
     }
 
     auto timestamp = api::max_timestamp;
-    auto memtable_min_timestamp = table_s.min_memtable_timestamp();
-    // Use memtable timestamp if it contains data older than the sstables being compacted,
+    api::timestamp_type memtable_min_timestamp;
+    if (is_shadowable) {
+        memtable_min_timestamp = table_s.min_memtable_live_row_marker_timestamp();
```

Comment:
What if all live row markers have a larger timestamp than the currently considered shadowable tombstone, but the row this tombstone covers has data in the memtable and that row doesn't have a live row marker?
Our current MV should always write a row marker, but can we guarantee that?

Reply:
I think the whole "shadowable tombstone" implementation and its bizarre logic relying on row markers depends completely on the MV logic of creating these row markers. It can never be used outside of MV.

Reply:
@nyh?